setup-gcloud/tests/download-util.test.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Tests download-util.
+ */
+import 'mocha';
+import { expect } from 'chai';
+
+import fs from 'fs';
+import os from 'os';
+import * as io from '@actions/io';
+
+import { TestToolCache, TEST_SDK_VERSION } from '../src/test-util';
+const [toolDir, tempDir] = TestToolCache.override();
+
+// Import modules being tested after test setup as run.
+import * as downloadUtil from '../src/download-util';
+
+import { getReleaseURL } from '../src/format-url';
+
+describe('#downloadAndExtractTool', function() {
+  beforeEach(async function() {
+    await io.rmRF(toolDir);
+    await io.rmRF(tempDir);
+  });
+
+  afterEach(async function() {
+    await io.rmRF(toolDir);
+    await io.rmRF(tempDir);
+  });
```

Comment:
Please be sure to preserve the original test timeouts, here and throughout. The MacOS runners can be very flaky if you don't

Reply:
Each test has a an individual timeout of 30s per the mocha command in the package.json. Do you want it to be higher?

Reply:
30s works. Thanks for clarifying :)